pallets/stellar-relay/src/lib.rs

```diff
@@ -630,7 +627,17 @@ pub mod pallet {
 
 				// Check if the externalized value is the same for all envelopes
 				ensure!(externalized_value == value, Error::<T>::ExternalizedValueMismatch);
-				ensure!(externalized_n_h == n_h, Error::<T>::ExternalizedNHMismatch);
+
+				// use this envelopes's n_h as basis for the comparison with the succeeding
+				// envelopes
+				if externalized_n_h == u32::MAX {
```

Comment:
If we want to be super strict, we would additionally test here and in line 638 that the message is also `ScpStExternalize` but I think the case that there is an `ScpStConfirm` message with `n_h` equal to `u32::MAX` will not occur (and this would also make the code more ugly).